### PR TITLE
feat(cli): add cmd to collect withdrawal events and submit as actions

### PR DESCRIPTION
**Pull request by SuperFluffy**
## Summary
Adds `bridge collect-withdrawals` and `bridge submit-withdrawals` subcommands to `astria-cli`.

## Background
The worker service `astria-bridge-withdrawer` is a closed system that collects withdrawal event from the rollup and submits them to the sequencer in a closed loop. But it can be desirable to inspect the withdrawal events generated by the bridge contracts, and then submit them manually. This functionality is now provided by `astria-cli`.

## Changes
- Add subcommands `bridge collect-withdrawal-s` to `astria-cli`, which has two different modes of operation, depending on whether `--to-rollup-height` is set:
    - if set, it fetches all blocks between `--from-rollup-height` and `--to-rollup-height` (inclusive), converts them to seuquencer actions, and then writes them to a file.
    - if not set, it fetches blocks from `--from-rollup-height` until SIGINT (Ctrl-C) is received. After the signal, the converted actions are written to a file.

## Testing
- Added `run-smoke-cli` in `charts/deploy.just` (@joroshiba)
- Added a `smoke-cli:` in `.github/workflows/docker-build.yaml` (@joroshiba)

### CODE REVIEW

File: crates/astria-cli/src/commands/bridge.rs
Comment (noot) [inline crates/astria-cli/src/commands/bridge.rs:66]: it might be nice to also have a `to_block` option instead of always using the current head block by default

File: crates/astria-cli/src/commands/bridge.rs
Comment (noot) [inline crates/astria-cli/src/commands/bridge.rs:122]: you don't use the subscription anywhere right? you can just get the latest block and use the number from that instead: https://docs.rs/ethers/latest/ethers/middleware/trait.Middleware.html#method.get_block

File: crates/astria-cli/src/commands/bridge.rs
Comment (noot) [inline crates/astria-cli/src/commands/bridge.rs:421]: use `Some(bridge_address)`

File: crates/astria-cli/src/commands/bridge.rs
Comment (noot) [inline crates/astria-cli/src/commands/bridge.rs:462]: also here: `Some(bridge_address)`

File: crates/astria-cli/src/commands/bridge.rs
Comment (noot) [inline crates/astria-cli/src/commands/bridge.rs:399]: so this is not applicable anymore - we need to also fix this in the withdrawer. this will return funds back to the sequencer address, but we want them to be refunded back to the rollup. we can leave this to a follow-up. basically this should be set to the sender of the withdrawal tx on the rollup, which you should be able to extract this from the transaction data.

File: crates/astria-cli/src/commands/bridge.rs
Comment (SuperFluffy) [reply crates/astria-cli/src/commands/bridge.rs:122]: The subscription gets used further down in the chained stream (not correct code, just for demonstration purposes):
```rust
let incoming_blocks = iter(start_height..current_hight).then(|height| providers.get_bock(height)),chain(block_subscription);
```
Doing it this way is import to not have the rollup advance between getting `current_height`, fetching blocks from `start` to `current_height`, and then going back to the stream.

From a first glance the ethers subscription buffers blocks so that we are not going to miss out

File: crates/astria-cli/src/commands/bridge.rs
Comment (SuperFluffy) [reply crates/astria-cli/src/commands/bridge.rs:399]: So this is actually rollup-specific data, not sequencer-specific?

Should we then make this a breaking change and coll this field `bytes rollup_return_address = x;` in the protobuf?

File: crates/astria-cli/src/commands/bridge.rs
Comment (SuperFluffy) [reply crates/astria-cli/src/commands/bridge.rs]: Done in  01da7859

File: crates/astria-cli/src/commands/bridge.rs
Comment (SuperFluffy) [reply crates/astria-cli/src/commands/bridge.rs]: Done in  01da7859

File: crates/astria-cli/src/commands/bridge.rs
Comment (SuperFluffy) [reply crates/astria-cli/src/commands/bridge.rs]: I have renamed this line to `from_rollup_height` and added an arg `to_rollup_height: Option<u64>` in [f513e17](https://github.com/astriaorg/astria/pull/1261/commits/f513e1783620d4e6cdfb54ca2111935a3f1685f1).

The stream is now bounded if `--to-rollup-height` is set, or unbounded otherwise (requiring ctrl-c to exit).

File: crates/astria-cli/src/commands/bridge.rs
Comment (SuperFluffy) [reply crates/astria-cli/src/commands/bridge.rs]: Done in [8333ecf](https://github.com/astriaorg/astria/pull/1261/commits/8333ecfad8bec2d48d213fd27234242e3f1638dc) in after merging https://github.com/astriaorg/astria/pull/1266

**Review by joroshiba — APPROVED** [review]

**Pull request merged**
